ui/tx/Noves/components/NovesActionCard.tsx

```diff
@@ -0,0 +1,114 @@
+import { Box, Hide, Popover, PopoverArrow, PopoverContent, PopoverTrigger, Show, Text, useColorModeValue } from '@chakra-ui/react';
+import type { FC } from 'react';
+import React from 'react';
+
+import lightning from 'icons/lightning.svg';
+import Icon from 'ui/shared/chakra/Icon';
+import TokenEntity from 'ui/shared/entities/token/TokenEntity';
+
+import type { NovesAction, NovesFlowViewItem } from '../utils/NovesGenerateFlowViewData';
+import NovesTokensCard from './NovesTokensCard';
+
+interface Props {
+  item: NovesFlowViewItem;
+}
+
+const NovesActionCard: FC<Props> = ({ item }) => {
+  const popoverBg = useColorModeValue('gray.700', 'gray.300');
+
+  const getTokenData = (action: NovesAction) => {
```

Comment:
I believe this doesn't need to be a function. It can be simplified to a value using the React.useMemo hook.